src/interfaces/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { History } from 'history';
+import { FormikValues } from 'formik';
+
+export interface IbeerObject{
+  beerName: string,
+  desc: string,
+  img: string,
+  rating: number
+}
+
+export interface Idata{
+  id: string,
+  data: IbeerObject
+}
+
+ export interface Iaction{
+  type: string,
+  payload: any
+}
+
+export interface IProps {
+  addNewBeer: (beer: FormikValues, img: File) => any
+  history: History;
+}
```

Comment:
You can't have global interface IProps, you dont know what component will use it.

Also this whole file should contain only GLOBAL interfaces, rest should be declared near the modules/domains they belong to.